### PR TITLE
WFLY-8770 Integrate aws.S3_PING discovery protocol

### PR DESCRIPTION
**Pull request by rhusar**
Resolves
https://issues.redhat.com/browse/WFLY-8770

Analysis document: 
https://github.com/wildfly/wildfly-proposals/pull/439

### CODE REVIEW

File: ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/amazon/aws/core/main/module.xml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+  ~ JBoss, Home of Professional Open Source.
+  ~ Copyright 2022, Red Hat Middleware LLC, and individual contributors
```
Comment (rhusar) [inline ee-feature-pack/common/src/main/resources/modules/system/layers/base/com/amazon/aws/core/main/module.xml:5]: Fix the stray legacy header.

**Comment by rhusar** [comment]
Just completed a manual pre-check (after this many rebases) against my live S3 bucket and it works like charm.

```
22:06:06,023 INFO  [org.jgroups.protocols.aws.S3_PING] (ServerService Thread Pool -- 90) using Amazon S3 ping in region eu-central-1 with bucket 'rhusar' and prefix ''
22:06:06,384 INFO  [org.jgroups.protocols.aws.S3_PING] (ServerService Thread Pool -- 90) found bucket rhusar
...
22:06:06,440 INFO  [org.jboss.as.clustering.jgroups] (ServerService Thread Pool -- 90) WFLYCLJG0032: Connecting 'ee' channel. 'ribera.docsis.vodafone.cz-2' joining cluster 'ejb' via /127.0.0.1:7700
...
22:06:07,448 INFO  [org.infinispan.CLUSTER] (ServerService Thread Pool -- 93) ISPN000078: Starting JGroups channel `ejb`
...
22:06:07,453 INFO  [org.infinispan.CLUSTER] (ServerService Thread Pool -- 94) ISPN000094: Received new cluster view for channel ejb: [ribera|1] (2) [ribera, ribera.docsis.vodafone.cz-2]
```

**Review by pferraro — APPROVED** [review]

File: boms/common-ee/pom.xml [hunk 2]
```diff
@@ -2386,6 +2458,18 @@
                 </exclusions>
             </dependency>
 
+            <dependency>
+                <groupId>software.amazon.ion</groupId>
+                <artifactId>ion-java</artifactId>
+                <version>${version.com.amazon.ion}</version>
```
Comment (bstansberry) [inline boms/common-ee/pom.xml:2464]: @rhusar FYI:

https://amzn.github.io/ion-docs/news/2020/01/15/software.amazon.ion-deprecated.html

This is just FYI, as this dep comes via com.amazonaws:aws-java-sdk-core:jar:1.12.284, but even the latest release of that still uses software.amazon.ion:

https://repo1.maven.org/maven2/com/amazonaws/aws-java-sdk-core/1.12.334/aws-java-sdk-core-1.12.334.pom

**Comment by bstansberry** [comment]
Thanks @rhusar

File: boms/common-ee/pom.xml (hunk 2 above)
Comment (rhusar) [reply boms/common-ee/pom.xml:2464]: Yup... they are a bit messy. I will keep an eye on that when they clean that up.

**Pull request merged**